src/utils/misc.cpp

```diff
@@ -1300,6 +1300,29 @@ bool Utils::Misc::isNoteEditingAllowed() {
     return SettingsService().value(QStringLiteral("allowNoteEditing"), true).toBool();
 }
 
+/**
+ * Returns if "hideIconsInMenus" is turned on
+ *
+ * @return
+ */
+bool Utils::Misc::areMenuIconsHidden() {
+#ifdef Q_OS_MAC
+    // hide icons in menus by default on macOS
+    bool hideIconsInMenusDefault = true;
+#else
+    bool hideIconsInMenusDefault = false;
+#endif
+
+    SettingsService settings;
+
+    if (!settings.contains(QStringLiteral("hideIconsInMenus"))) {
```

Comment:
You don't need that whole if-clause, because there is no old setting to migrate from.